microapidev/comment-microapi
# Worked on issue #52 controller to handle post requests for adding comments

**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Summarise the main tasks handled in the pull request
I added a route and controller to handle post requests for adding comments
**description of Task to be completed?**
Controller to handle post requests for adding comments
Describe (in detail) what the task completed in the pull request does as per the relevant task
When a post request is sent to the /comments endpoint the controller will find the id of the user from the db using the commentOwnerEmail in the req body and pass it to the commentOwner field before saving the comment.If the user commenting doesn't exist in the db yet, the user is saved using the commentOwnerName and commentOwnerEmail in the request body then the id of the newly saved user is pulled from the db and passed to the commentOwner field and after that is done the comment is saved in the database.
**How should this be manually tested?**
Make a post request to the /comments endpoint and make sure the body of the comment in the documentation.
Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

-clone the repo

-Install dependencies `npm install`

-Run the app `npm run startDev`

-Open postman and send a post request to localhost:4000/comments,make sure the request body is as stated in the documentation.

You should get a response containing the comment you sent as it appears in the db.
**Any background context you want to provide?**
Test request body
![Screenshot from 2020-06-29 00-47-27](https://user-images.githubusercontent.com/29974159/85959391-deaee380-b9a4-11ea-8eef-0c53b534a823.png)
Response
![Screenshot from 2020-06-29 00-47-41](https://user-images.githubusercontent.com/29974159/85959395-ea020f00-b9a4-11ea-9c37-8ccaf0df4f8e.png)

Any pertinent information that should be considered

**What is the link to the issue on Github?**
https://github.com/microapidev/comment-microapi/issues/52
[Issue #14](https://github.com/microapidev/comment-microapi/issues/14)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here
